desc/protoparse/parser.go

```diff
@@ -104,6 +104,18 @@ type Parser struct {
 	// with import statements all being relative to the root of this tree).
 	InferImportPaths bool
 
+	// DependencyDescriptors is a map of filenames to file descriptor protos,
+	// which will be consulted when resolving imports. This allows a compiled Go
+	// proto in another Go module to be referenced in the proto(s) being parsed.
+	//
+	// In the event of a filename collision, Accessor is consulted first,
+	// then FileDescriptors is consulted, and finally the well-known protos
+	// are used.
+	//
+	// In order to get descriptors from compiled Go protos, use the
+	// `desc.LoadFileDescriptor` method.
+	DependencyDescriptors map[string]*desc.FileDescriptor
```

Comment:
Do these really need to be `*desc.FileDescriptor`? I would think it should suffice to instead supply `*descriptor.FileDescriptorProto` instances (such as would be found in a file descriptor set created by protoc).

Reply:
That was what I originally used, back when I was including another helper method. However, `LoadFileDescriptor` (your method in `desc`) returns a `*desc.FileDescriptor`, so I would be asking the user to do _more_ work to get that and then call `.AsFileDescriptorProto()`.

Ideally, I would like to accept the thing that it is easiest for the user to get.

Reply:
That said, with your other idea, this should definitely stay as is.

Reply:
This should be good now. I did do a function receiver like you suggested -- that seems like a really good idea to me.